From 96df076d4b077ae478bbfd1283059c13e34b25bf Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Sun, 9 Dec 2018 13:05:52 -0700
Subject: [PATCH 43/48] Revert "SystemUI: add navbar layout inversion tuning"

* This was accidentally merged

This reverts commit eb57db3ef24ebb03ac2ea1909c748a686f386d1c.

Change-Id: I74c0d8408f65d7cfe8c7a412070a4fcda398c7f0

# Conflicts:
#	packages/SystemUI/res/values/cm_strings.xml
---
 packages/SystemUI/res/xml/nav_bar_tuner.xml   |  4 ---
 .../phone/NavigationBarInflaterView.java      | 34 +------------------
 2 files changed, 1 insertion(+), 37 deletions(-)

diff --git a/packages/SystemUI/res/xml/nav_bar_tuner.xml b/packages/SystemUI/res/xml/nav_bar_tuner.xml
index a560fa4307b..68e8fad1e24 100644
--- a/packages/SystemUI/res/xml/nav_bar_tuner.xml
+++ b/packages/SystemUI/res/xml/nav_bar_tuner.xml
@@ -64,8 +64,4 @@
         android:summary="%s"
         android:title="@string/right_icon" />
 
-    <com.android.systemui.tuner.TunerSwitch
-        android:key="sysui_nav_bar_inverse"
-        android:title="@string/nav_bar_layout_inverse" />
-
 </PreferenceScreen>
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/NavigationBarInflaterView.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/NavigationBarInflaterView.java
index 24b729141cf..35ec3325aee 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/NavigationBarInflaterView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/NavigationBarInflaterView.java
@@ -45,8 +45,6 @@ import com.android.systemui.tuner.TunerService;
 import com.android.systemui.tuner.TunerService.Tunable;
 
 import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
 
@@ -60,7 +58,6 @@ public class NavigationBarInflaterView extends FrameLayout
     public static final String NAV_BAR_VIEWS = "sysui_nav_bar";
     public static final String NAV_BAR_LEFT = "sysui_nav_bar_left";
     public static final String NAV_BAR_RIGHT = "sysui_nav_bar_right";
-    public static final String NAV_BAR_INVERSE = "sysui_nav_bar_inverse";
 
     public static final String MENU_IME_ROTATE = "menu_ime";
     public static final String BACK = "back";
@@ -106,8 +103,6 @@ public class NavigationBarInflaterView extends FrameLayout
 
     private OverviewProxyService mOverviewProxyService;
 
-    private boolean mInverseLayout;
-
     public NavigationBarInflaterView(Context context, AttributeSet attrs) {
         super(context, attrs);
         createInflaters();
@@ -157,7 +152,7 @@ public class NavigationBarInflaterView extends FrameLayout
     protected void onAttachedToWindow() {
         super.onAttachedToWindow();
         Dependency.get(TunerService.class).addTunable(this, NAV_BAR_VIEWS, NAV_BAR_LEFT,
-                NAV_BAR_RIGHT, NAV_BAR_INVERSE);
+                NAV_BAR_RIGHT);
         Dependency.get(PluginManager.class).addPluginListener(this,
                 NavBarButtonProvider.class, true /* Allow multiple */);
     }
@@ -180,10 +175,6 @@ public class NavigationBarInflaterView extends FrameLayout
         } else if (NAV_BAR_LEFT.equals(key) || NAV_BAR_RIGHT.equals(key)) {
             clearViews();
             inflateLayout(mCurrentLayout);
-        } else if (NAV_BAR_INVERSE.equals(key)) {
-            mInverseLayout = newValue != null && Integer.parseInt(newValue) != 0;
-            clearViews();
-            inflateLayout(mCurrentLayout);
         }
     }
 
@@ -260,17 +251,6 @@ public class NavigationBarInflaterView extends FrameLayout
         }
     }
 
-    private String[] swapLeftAndRight(String[] set) {
-        for (int i = 0; i < set.length; i++) {
-            if (set.equals(LEFT)) {
-                set[i] = RIGHT;
-            } else if (set[i].equals(RIGHT)) {
-                set[i] = LEFT;
-            }
-        }
-        return set;
-    }
-
     protected void inflateLayout(String newLayout) {
         mCurrentLayout = newLayout;
         if (newLayout == null) {
@@ -285,18 +265,6 @@ public class NavigationBarInflaterView extends FrameLayout
         String[] start = sets[0].split(BUTTON_SEPARATOR);
         String[] center = sets[1].split(BUTTON_SEPARATOR);
         String[] end = sets[2].split(BUTTON_SEPARATOR);
-        // Invert start, center and end if needed.
-        if (mInverseLayout) {
-            List<String> newStart = Arrays.asList(end);
-            List<String> newCenter = Arrays.asList(center);
-            List<String> newEnd = Arrays.asList(start);
-            Collections.reverse(newStart);
-            Collections.reverse(newCenter);
-            Collections.reverse(newEnd);
-            start = swapLeftAndRight((String[]) newStart.toArray());
-            center = swapLeftAndRight((String[]) newCenter.toArray());
-            end = swapLeftAndRight((String[]) newEnd.toArray());
-        }
         // Inflate these in start to end order or accessibility traversal will be messed up.
         inflateButtons(start, mRot0.findViewById(R.id.ends_group), isRot0Landscape, true);
         inflateButtons(start, mRot90.findViewById(R.id.ends_group), !isRot0Landscape, true);
-- 
2.17.1

